### ISSUE TITLE
Background color of the selected text in the Extensions panel and Settings page is almost indistinguishable

### ISSUE THREAD

**Issue by Z-E-D** [microsoft/vscode#145462]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.65.0
- OS Version: Windows 7

Steps to Reproduce:

1. choose Default High Contrast theme
2. open the Extensions panel
3. type any text in the text box, e.g. "javascript"
4. select a part of that text -> the background color is very dark and hardly indistinguishable from the black background of the text box.

The same problem is with the selected text in the top text box on the Settings page. However, the text boxes in the Search panel and Find widget have different and more prominent background color of the selected text.

This is not problem only with that theme, there are others with the same problem:
Kimbie Dark, Monokai, Monokai Classic, Github Dark Colorblind/Default/Dimmed, One Dark Pro...

I tried to override the background color of the selected text with the Default High Contrast theme that I am using with:
"selection.background": "#ff0000"

but the background color of selected text in Extensions/Settings is still much more darker than that in the Search/Find text boxes.

**Comment by sandy081** [comment]
Both these components use `SuggestEnabledInput` and not sure who owns this widget.

CC @alexdima

**Comment by Z-E-D** [comment]
Dark High Contrast:
Extensions:
![Dark High Contrast - Extensions](https://user-images.githubusercontent.com/97363660/159263373-0944869b-88bb-4806-a757-e22c4c0e0eec.png)
Settings:
![Dark High Contrast - Settings](https://user-images.githubusercontent.com/97363660/159263365-a195605b-846d-4b9d-b392-8a5cdb703ef0.png)
Search:
![Dark High Contrast - Search](https://user-images.githubusercontent.com/97363660/159263369-396013a9-a6fd-4c1a-a7d3-894b3363846a.png)
Find:
![Dark High Contrast - Find](https://user-images.githubusercontent.com/97363660/159263366-cf4ea1b1-4e90-4f71-8e2b-7e28ecf29489.png)

`"selection.background": "#008000"`

This theme is somewhat better, but still has different selected background colors, with Extensions/Settings being much darker than Search/Find:
Red:
Extensions:
![Red - Extensions](https://user-images.githubusercontent.com/97363660/159268981-83f70ce5-a50d-4283-8a87-68f436efc925.png)
Settings:
![Red - Settings](https://user-images.githubusercontent.com/97363660/159268979-82f3635b-ad13-4407-9d38-c1e830836455.png)
Search:
![Red - Search](https://user-images.githubusercontent.com/97363660/159268976-eeef81a1-8c85-4cf9-b93c-532bad5f5859.png)
Find:
![Red - Find](https://user-images.githubusercontent.com/97363660/159268982-2e7feefd-35c1-4dc1-b836-b095e46c153a.png)
`"selection.background": "#ff777788",`

This theme is strange one since it has different selected background colors, not only darker:
Abyss:
Extensions:
![Abyss - Extensions](https://user-images.githubusercontent.com/97363660/159264470-8d386b0e-4e38-47a2-bab8-6d66e57a5fee.png)
Settings:
![Abyss - Settings](https://user-images.githubusercontent.com/97363660/159264473-f70e6af8-e023-4e87-b0d2-3157c5fc38fe.png)
Search:
![Abyss - Search](https://user-images.githubusercontent.com/97363660/159264478-533a1fc9-9cde-4f5d-9583-632ce8661852.png)
Find:
![Abyss - Find](https://user-images.githubusercontent.com/97363660/159264477-1abee10a-09d2-435e-b7d6-c6c7d7327c5b.png)
It doesn't have "selection.background" in its theme's json. Its Search/Find color is "#0562d5" which is nowhere found in its .json, so I suppose it is a default one, but the Extensions/Settings has the selected background color which is the same as:
`"editor.selectionBackground": "#770811,"`

Why should Extensions/Settings selected background color have the same color as the selected background text in the editor? Shouldn't it be the same as the selected background color of the Search/Find text boxes? If theme doesn't have defined "selection.background", the Extensions/Settings should have default color as Search/Find, not the color from editor.

**Comment by roblourens** [comment]
This is discussed in a few old issues, like https://github.com/microsoft/vscode/issues/75012. Basically we are trying to mimic a native text input with an editor control for some inputs that use editor features. But we can't get it perfect. The entirely different colors come from the fact that the blue is a platform-specific color, it's completely different on Mac, and the editor doesn't use that color.

**Comment by Z-E-D** [comment]
I don't want to pretend that I understand how VSCode internally works. I am just curious, if you don't mind to explain, why Extension/Settings text boxes are not created as Search/Find boxes with a native text input? I would expect that Search/Find would require editor features more than Extensions/Settings, not otherwise, but I am just beginner with VSCode, so please forgive my ignorance.

Anyway, within the function registerThemingParticipant in the file vs\workbench\contrib\codeEditor\browser\suggestEnabledInput\suggestEnabledInput.ts I changed:
`selectionColor.transparent(0.4)`
to:
`selectionColor.transparent(1)`
and now the selected background color of the Extensions/Settings is much more prominent. When the theme has defined "editor.selectionBackground", it would have the same colors in Extensions/Settings as Search/Replace (i.e. they are not darker). However, when the theme doesn't have that setting defined, then the selected background color will be the same as the selected background of the text in editor. Which is not perfect, but it is better than before, IMHO.

Could you please check my modification. As I said, I am just starting with VSCode, so maybe I am doing something wrong.

**Comment by roblourens** [comment]
It uses an editor because it has a suggest widget pop up to help with some queries.

From my memory, that might work with some themes/color combinations, but not others. I picked 0.4 as a compromise. I haven't looked at this in a long time though, maybe a different number will be better now. This is also different across OS as I recall.

**Comment by Z-E-D** [comment]
The modification works fine with all themes that I tried. I don't understand what OS has with it since it is applied to the "editor.selectionBackground" color defined by theme.

Also, I don't see the point to apply transparent(0.4) to the background color if it defined by theme, but not to apply it if it is not defined by theme in which case it has the same value as the editor selected background color.

**Comment by roblourens** [comment]
I think that we resolved or improved this at some point
